pydicom/config.py

```diff
@@ -152,15 +153,93 @@ def DS_decimal(use_Decimal_boolean: bool = True) -> None:
 Default ``False``.
 """
 
+
 enforce_valid_values = False
-"""Raise exceptions if any value is not allowed by DICOM Standard.
+"""Deprecated.
+Use :attr:`Settings.reading_validation_mode` instead.
+"""
 
-e.g. DS strings that are longer than 16 characters; IS strings outside
-the allowed range.
 
-Default ``False``.
+# Constants used to define how data element values shall be validated
+NO_CHECK = 0
+"""If one of the validation modes is set this value, no value validation
+will be performed.
+"""
+
+WARN = 1
+"""If one of the validation modes is set to this value, a warning is issued if
+a value validation error occurs.
+"""
+
+RAISE = 2
+"""If one of the validation modes is set to this value, an exception is raised
+if a value validation error occurs.
+"""
+
+
+class Settings:
+    """Collection of several configuration values.
+    Accessed via the singleton :attr:`settings`.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    reading_validation_mode : int
+        Defines behavior for value validation while reading value.
+        Value validation checks if a value is allowed by the DICOM Standard,
+        e.g. that DS strings are not longer than 16 characters and contain only
```

Comment:
Wording is a little awkward here.  Perhaps "Defines behavior of validation while reading values, compared with the DICOM standard.  e.g. that DS ..."

Reply:
Agreed.